### PR TITLE
Updating documentation for ECC server certificate chain support.

### PR DESCRIPTION
**Pull request by CIPop**
Documentation to support IoT Hub Preview ECC certificate chain (https://docs.microsoft.com/azure/iot-hub/iot-hub-tls-support#elliptic-curve-cryptography-ecc-server-tls-certificate-preview)

### CODE REVIEW

File: certs/readme.md [hunk 1]
```diff
@@ -33,3 +33,34 @@ A couple of examples:
 ## Additional Information
 
 For additional guidance and important information about certificates, please refer to [this blog post](https://techcommunity.microsoft.com/t5/internet-of-things/azure-iot-tls-critical-changes-are-almost-here-and-why-you/ba-p/2393169) from the security team.
+
```
Comment (danewalton) [inline certs/readme.md:36]: nit: do we need bearssl?

**Review by danewalton — APPROVED** [review]

File: certs/readme.md [hunk 2]
```diff
@@ -33,3 +33,34 @@ A couple of examples:
...
+## IoT Hub ECC Root
+
+To work with the new Azure Cloud ECC server certificate chain, the TLS stack must be configured to prevent RSA cipher-suites from being advertised as described [here](https://docs.microsoft.com/azure/iot-hub/iot-hub-tls-support#elliptic-curve-cryptography-ecc-server-tls-certificate-preview).
```
Comment (momuno) [inline certs/readme.md:39]: nit: extra space in front?

File: certs/readme.md [hunk 3]
```diff
@@ -33,3 +33,34 @@ A couple of examples:
 ## Additional Information
 
 For additional guidance and important information about certificates, please refer to [this blog post](https://techcommunity.microsoft.com/t5/internet-of-things/azure-iot-tls-critical-changes-are-almost-here-and-why-you/ba-p/2393169) from the security team.
```
Comment (momuno) [inline certs/readme.md:35]: nit: extra space in front?

File: certs/readme.md [hunk 4]
```diff
@@ -33,3 +33,34 @@ A couple of examples:
...
+To work with the new Azure Cloud ECC server certificate chain, the TLS stack must be configured to prevent RSA cipher-suites from being advertised as described [here](https://docs.microsoft.com/azure/iot-hub/iot-hub-tls-support#elliptic-curve-cryptography-ecc-server-tls-certificate-preview).
+
+### Using OpenSSL
+
+Use the following option to limit the advertised cipher-suites:
+```C
+IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_OPENSSL_CIPHER_SUITE, "ECDH+ECDSA+HIGH");
+```
+
+### Using WolfSSL
```
Comment (momuno) [inline certs/readme.md:48]: I believe wolfSSL is lowercase `w`.  https://www.wolfssl.com/

File: certs/readme.md [hunk 5]
```diff
@@ -33,3 +33,34 @@ A couple of examples:
...
+
+Rebuild WolfSSL without RSA support by adding `--disable-rsa` to the `configure` step.
```
Comment (momuno) [inline certs/readme.md:50]: same here.

File: iothub_client/devdoc/iothubclient_c_library.md [hunk 6]
```diff
@@ -501,6 +501,13 @@ const char* x509privatekey =
 ```
 
 - "CypherSuite" - only available when OpenSSL is used. value is a pointer to a null terminated string that contains a list in the format specified by [SSL_set_cipher_list](https://www.openssl.org/docs/man1.1.1/man3/SSL_CTX_set_cipher_list.html).
+
```
Comment (momuno) [inline iothub_client/devdoc/iothubclient_c_library.md:504]: I'm not following the flow here.  Are "Engine" and "x509PrivateKeyType" other OPTION_OPENSSL_CIPER_SUITE values??  If not, perhaps they need their own header? Not sure.

**Review by momuno — APPROVED** [review]

File: certs/readme.md (hunk 1 above)
Comment (CIPop) [reply certs/readme.md:36]: Good point. we won't be supporting that (except if Hub decides to prefer ECC).
1. I can't find an easy way to disable RSA during compilation of BearSSL.
2. We don't allow ciphersuite selection today. We call `br_ssl_client_init_full` which is using all ciphersuites (recommended if size is not an issue).

File: iothub_client/devdoc/iothubclient_c_library.md (hunk 6 above)
Comment (CIPop) [reply iothub_client/devdoc/iothubclient_c_library.md:504]: Good catch! Fixed.

**Pull request merged**